best_practices.md

```diff
@@ -0,0 +1,271 @@
+# Best practices for programming diagnostics
+
+To facilitate code integration, code sharing between developers, reusability, easy
```

Comment:
These properties are objectives for tool(s) stakeholders, but not necessarily for scripts contributors, except indirectly. So these recommendations, which are anyway very complete and up-to-date,  are definitely shaping contributions toward a given philosophy of cooperation which may be desirable, but may be also be considered as putting too much constraints, and then discourage contributors. This applies both to openness recommendation (licence) and to other, technical recommendations which, taken all together, represent a much significant technical step for the average scientist.
So, is it time to enforce such best practices (even at the level of a recommendation) ? My understanding is that this better fits as a goal for ESMValTool contributors than as a part of an agnostic standard.

Reply:
Transferred to #7.